ktor-http/common/src/io/ktor/http/URLParser.kt

```diff
@@ -41,6 +41,30 @@ internal fun URLBuilder.takeFromUnsafe(urlString: String): URLBuilder {
     val slashCount = count(urlString, startIndex, endIndex, '/')
     startIndex += slashCount
 
+    if (protocol.name == "file") {
+        require(slashCount == 3)
```

Comment:
A message is required: we don't support host in file URIs so we need to clearly state it

Reply:
Thanks for the notice, I will add support and tests